src/plugins/mce/plugin.c

```diff
@@ -78,24 +103,24 @@ mce_signal_filter (NCore *core, DBusConnection *connection, DBusMessage *msg, vo
     (void) connection;
     (void) user_data;
 
-    if (dbus_message_is_signal(msg, MCE_SIGNAL_IF, MCE_LED_PATTERN_DEACTIVATED_SIG)) {
+    if (dbus_message_is_signal (msg, MCE_SIGNAL_IF, MCE_LED_PATTERN_DEACTIVATED_SIG)) {
         DBusError error;
-        dbus_error_init(&error);
+        dbus_error_init (&error);
 
         gchar *pattern = NULL;
 
-        if (!dbus_message_get_args(msg, &error, DBUS_TYPE_STRING, &pattern, DBUS_TYPE_INVALID)) {
+        if (!dbus_message_get_args (msg, &error, DBUS_TYPE_STRING, &pattern, DBUS_TYPE_INVALID)) {
             N_WARNING (LOG_CAT "%s >> failed to read MCE signal arguments, cause: %s", __FUNCTION__, error.message);
-            dbus_error_free(&error);
+            dbus_error_free (&error);
         } else {
             GList *event;
             N_DEBUG (LOG_CAT "%s >> mce finished playing %s", __FUNCTION__, pattern);
-            for (event = active_events; event != NULL; event = g_list_next(event)) {
+            for (event = active_events; event != NULL; event = g_list_next (event)) {
                 MceData *data = (MceData *) event->data;
-                if (g_strcmp0(pattern, data->pattern) == 0) {
-                    n_sink_interface_complete(data->iface, data->request);
+                if (g_strcmp0 (pattern, data->pattern) == 0 && data->playing) {
+                    active_events = g_list_remove_all (active_events, data);
```

Comment:
I think you should remove the data from the list, even if the playing state is FALSE.

Reply:
If done so, then after the MCE_ACTIVATE_LED_PATTERN call finishes, the active_events list wouldn't contain the pattern, but the pattern would be still active on mce side.

Reply:
...and the next stop() call here would skip sending the DEACTIVATE message to mce and the pattern would remain on.